Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -457,29 +389,53 @@ public void E2E_ExcludedProjects()
   <Compile Include='{code1}' />
 </ItemGroup>
 ";
-            var projectFilePath = CreateProjectFile(projectXml, rootOutputFolder, rootConfigFolder);
+            var projectFilePath = context.CreateProjectFile(projectXml);
 
             // Act
             var result = BuildRunner.BuildTargets(TestContext, projectFilePath);
 
             // Assert
             result.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget); // Build should succeed with warnings
-            var projectSpecificOutputDir = CheckProjectSpecificOutputStructure(rootOutputFolder);
 
-            var actualProjectInfo = CheckProjectInfoExists(projectSpecificOutputDir);
-            actualProjectInfo.IsExcluded.Should().BeTrue();
+            var actualStructure = context.CheckProjectSpecificStructure(checkConfigFiles: false);
+            actualStructure.ProjectInfo.IsExcluded.Should().BeTrue();
+            actualStructure.AssertConfigFileDoesNotExist(ExpectedProjectConfigFileName);
+            actualStructure.CheckExpectedFileList("\\code1.txt");
+        }
 
-            // This list of files should also be written
-            var projectSpecificConfigDir = CheckProjectSpecificConfigStructure(rootConfigFolder);
-            var expectedFilesToAnalyzeFilePath = AssertFileExists(projectSpecificConfigDir, ExpectedAnalysisFilesListFileName);
-            var fileList = File.ReadLines(expectedFilesToAnalyzeFilePath);
-            fileList.Should().BeEquivalentTo(new string[]
-            {
-                rootInputFolder + "\\code1.txt"
-            });
+        [TestMethod]
+        [TestCategory("E2E"), TestCategory("Targets")]
+        public void E2E_TestProjects()
+        {
+            // Project info and config should be written for files with $(SonarQubeTestProject) set to true
+            // Arrange
+            var context = CreateContext();
 
-            var actualFilesToAnalyze = actualProjectInfo.AssertAnalysisResultExists("FilesToAnalyze");
-            actualFilesToAnalyze.Location.Should().Be(expectedFilesToAnalyzeFilePath);
+            // Mix of analyzable and non-analyzable files
+            var foo1 = context.CreateInputFile("foo1.txt");
+            var code1 = context.CreateInputFile("code1.txt");
+            var projectXml = $@"
+<PropertyGroup>
+  <SonarQubeTestProject>true</SonarQubeTestProject>
```

Comment:
can you also add a test with an Explicit `<SonarQubeTestProject>false`, and verify 
```
actualStructure.ProjectConfig.ProjectType.Should().Be(ProjectType.Product);
```
?